csrc/executor.h

```diff
@@ -74,6 +74,9 @@ class TORCH_CUDA_CU_API FusionExecutor : public NonCopyable {
       const KernelArgumentHolder& args,
       const LaunchParams& launch_constraints);
 
+  //! To compile a fusion with the 32-bit index type, CompileParams
+  //! must be passed in. The index type of KernelArgumentHolder is no
+  //! longer used.
```

Comment:
Would it be possible to remove index type from `KernelArgumentHolder`?

Reply:
If I understand correctly, the only place that you strictly require index type for args is in `runFusion` because this is where you create structs for using as arguments of `cuLaunchKernel`. So can we move this logic
```C++
  // Replace unresolved TensorArgAbstract with the correct index type
  for (auto& arg : arguments_) {
    if (auto tensor_arg = dynamic_cast<TensorArgAbstract*>(arg.get())) {
      arg = getTensorArg(index_type, tensor_arg->getTensor());
    }
  }
```
into `runFusion` to avoid storing redundant information in `KernelArgumentHolder`?

Reply:
Yes, I think we can remove the index type from `KernelArgumentHolder` entirely. I remember I thought about it but don't remember why I didn't do so.

I'd think that logic could stay in `KernelArgumentHolder`, but only in `getBuffer()`. We could change `getBuffer` to have the index type as its parameter, and then it could swap the unresolved args.

Reply:
Done.

Reply:
I think this comment is no longer relevant?

Reply:
The first sentence is still true, but the second one should be updated as there's no index type at all.

Reply:
Updated